doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md

```diff
@@ -199,53 +222,123 @@ The DHCP_SERVER_IPV4_PORT table includes the following:
+#### NPU ####
 
-After the transformations performed by the interface-config service the interfaces configuration will have the following entries:
+1. systemd-sonic-generator renders configuration file of systemd-networkd and midplane-network.service according to the platform.json.
 
+``` text
+# bridge-midplane.netdev
+[NetDev]
+Name=bridge-midplane
+Kind=bridge
 ```
-auto bridge_midplane
-iface bridge_midplane inet static
-    bridge_ports Dpu0 Dpu1
-    address 169.254.200.254
-    netmask 255.255.255.0
-    network 169.254.200.0
-    broadcast 169.254.200.255
+
+``` text
+# bridge-midplane.network
+[Match]
+Name=bridge-midplane
+
+[Network]
+Address=169.254.200.254/24
 ```
 
-### IP assignment flow ###
+``` text
+# midplane-network.network
+[Match]
+Name=dpu0 dpu1
 
-![image](./smart-switch-ip-assignment-ip-assignment-flow.png)
+[Network]
+Bridge=bridge-midplane
+```
 
-- (1) sonic-cfggen based on hwsku.json renders midplane bridge and DHCP server configuration.
-- (2) sonic-cfggen pushes configuration into the Config DB.
-- (3-4) DHCP server and DHCP relay containers upon start consume configuration from the config DB and start listening for a requests from DHCP clients.
-- (5) DHCP client on the DPU sends a request over the eth0 interfaces. The request comes to the DHCP relay through the midplane bridge. DHCP relay inserts option 82 into the request with the information about the interface from which the request came. DHCP relay forwards the packet to the DHCP server.
-- (6) The DHCP server sends a reply with the IP configuration to the DHCP client on the DPU.
+``` text
+# database.service
+
+[Unit]
+...
+Requires=midplane-network.service
+After=midplane-network.service
+
+...
+```
+
+2. systemd-networkd helps to create the bridge-midplane interface and assign the specific IP address according to above configuration. Meanwhile, systemd-networkd will monitor the DPU PCIe interface. Once the PCIe interface is created, it will automatically add it into the bridge-midplane.
+3. midplane-network.service will be used to wait for the midplane bridge configured.
 
-### DPU ###
+``` text
+# midplane-network.service
 
-In the Smart Switch DPU, the PCIe interface should be used as a default management interface for communication with the switch and the outside world.
+[Unit]
+Description=Midplane network service
+Requires=systemd-networkd.service
+After=systemd-networkd.service
 
-It is the vendors' responsibility to make sure that the PCIe interface that is connected to the switch has given the `eth0` name.
+[Service]
+Type=oneshot
+User=root
+ExecStart=/usr/lib/systemd/systemd-networkd-wait-online -i bridge-midplane
 
-The IP address assignment for the default `eth0` management interface in the existing SONiC implementation is the following:
+[Install]
+WantedBy=multi-user.target
+```
+
+``` text
+# database.service
+
+[Unit]
+...
+Requires=midplane-network.service
+After=midplane-network.service
```

Comment:
It is not required to modify the `database.service`.
To make sure that `midplane-network.service` runs before `database.service` it is possible to use the `Before` section:
```
[Unit]
Description=Midplane network service
Requires=systemd-networkd.service
After=systemd-networkd.service
Before=database.service

[Service]
Type=oneshot
User=root
ExecStart=/usr/lib/systemd/systemd-networkd-wait-online -i bridge-midplane
```
This will allow us to make the `midplane-network.service` atomic and disable or remove it without affecting the `database.service`

Reply:
@oleksandrivantsiv I think both midplane-network and database services can come up independently without any dependency. @Pterosaur Can you explain why we need explicit dependency?

Reply:
> It is not required to modify the `database.service`. To make sure that `midplane-network.service` runs before `database.service` it is possible to use the `Before` section:
> 
> ```
> [Unit]
> Description=Midplane network service
> Requires=systemd-networkd.service
> After=systemd-networkd.service
> Before=database.service
> 
> [Service]
> Type=oneshot
> User=root
> ExecStart=/usr/lib/systemd/systemd-networkd-wait-online -i bridge-midplane
> ```
> 
> This will allow us to make the `midplane-network.service` atomic and disable or remove it without affecting the `database.service`

Done

Reply:
> @oleksandrivantsiv I think both midplane-network and database services can come up independently without any dependency. @Pterosaur Can you explain why we need explicit dependency?

1. To the DPU side, I believe I don't need to explain much detail. Because we have some components, including database.service, need to leverage the midplane network IP address to identify its DPU ID.
2. To the NPU side, I feel we need to bind the DPU Redis instance on the midplane bridge. If that IP is assigned after the Redis instance, I don't know whether the redis can delay bounding. Even if we suppose the redis support the delay bounding, but based on the current implementation, other components, like swsscommon, cannot support that. https://github.com/sonic-net/sonic-swss-common/blob/253ceb61916fabbf8f261aa9c7507e33d8628312/common/dbconnector.cpp#L536-L550 , So, other critical service may be crashed.